test/meta.cpp

```diff
@@ -141,10 +141,12 @@ static_assert(std::is_same<reverse_find<L, int>, list<int, float>>::value, "");
 
 struct check_integral
 {
+    constexpr check_integral() {}
     template <class T>
-    void operator()(T &&)
+    constexpr T operator()(T && i) const
     {
         static_assert(std::is_integral<T>{}, "");
+        return i;
```

Comment:
Why does the `check_integral` need a default constructor, and why does its function call operator need to return non-void?